controlplane/kubeadm/internal/controllers/helpers_test.go

```diff
@@ -522,140 +509,222 @@ func TestKubeadmControlPlaneReconciler_computeDesiredMachine(t *testing.T) {
-		}
-		g.Expect(createdMachine.Name).To(HavePrefix(kcp.Name))
-		g.Expect(createdMachine.Namespace).To(Equal(kcp.Namespace))
-		g.Expect(createdMachine.OwnerReferences).To(HaveLen(1))
-		g.Expect(createdMachine.OwnerReferences).To(ContainElement(*metav1.NewControllerRef(kcp, controlplanev1.GroupVersion.WithKind("KubeadmControlPlane"))))
-		g.Expect(createdMachine.Spec).To(BeComparableTo(expectedMachineSpec))
-
-		// Verify that the machineTemplate.ObjectMeta has been propagated to the Machine.
-		// Verify labels.
-		expectedLabels := map[string]string{}
-		for k, v := range kcpMachineTemplateObjectMeta.Labels {
-			expectedLabels[k] = v
-		}
-		expectedLabels[clusterv1.ClusterNameLabel] = cluster.Name
-		expectedLabels[clusterv1.MachineControlPlaneLabel] = ""
-		expectedLabels[clusterv1.MachineControlPlaneNameLabel] = kcp.Name
-		g.Expect(createdMachine.Labels).To(Equal(expectedLabels))
-		// Verify annotations.
-		expectedAnnotations := map[string]string{}
-		for k, v := range kcpMachineTemplateObjectMeta.Annotations {
-			expectedAnnotations[k] = v
-		}
-		expectedAnnotations[controlplanev1.KubeadmClusterConfigurationAnnotation] = clusterConfigurationString
-		// The pre-terminate annotation should always be added
-		expectedAnnotations[controlplanev1.PreTerminateHookCleanupAnnotation] = ""
-		g.Expect(createdMachine.Annotations).To(Equal(expectedAnnotations))
-
-		// Verify that machineTemplate.ObjectMeta in KCP has not been modified.
-		g.Expect(kcp.Spec.MachineTemplate.ObjectMeta.Labels).To(Equal(kcpMachineTemplateObjectMetaCopy.Labels))
-		g.Expect(kcp.Spec.MachineTemplate.ObjectMeta.Annotations).To(Equal(kcpMachineTemplateObjectMetaCopy.Annotations))
-	})
-
-	t.Run("should return the correct Machine object when updating an existing Machine", func(t *testing.T) {
-		g := NewWithT(t)
-
-		machineName := "existing-machine"
-		machineUID := types.UID("abc-123-existing-machine")
-		// Use different ClusterConfiguration string than the information present in KCP
-		// to verify that for an existing machine we do not override this information.
-		existingClusterConfigurationString := "existing-cluster-configuration-information"
-		remediationData := "remediation-data"
-		failureDomain := ptr.To("fd-1")
-		machineVersion := ptr.To("v1.25.3")
-		existingMachine := &clusterv1.Machine{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: machineName,
-				UID:  machineUID,
-				Annotations: map[string]string{
-					controlplanev1.KubeadmClusterConfigurationAnnotation: existingClusterConfigurationString,
-					controlplanev1.RemediationForAnnotation:              remediationData,
+	tests := []struct {
+		name                      string
+		kcp                       *controlplanev1.KubeadmControlPlane
+		isUpdatingExistingMachine bool
+		want                      []gomegatypes.GomegaMatcher
+	}{
+		{
+			name: "should return the correct Machine object when creating a new Machine",
```

Comment:
Let's add a test case for the default behaviour (no template) and for when the generated name exceeds the lenght

Reply:
done, following cases are added:

- User-defined template
- template exceeding 63 chars.
- invalid template
- no template

We are already testing some cases here: https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/topology/names/names_test.go